vowpalwabbit/queue.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <queue>
+
+// Mutex and CV cannot be used in managed C++, tell the compiler that this is unmanaged even if included in a managed
+// project.
+#ifdef _M_CEE
+#pragma managed(push, off)
+#undef _M_CEE
+#include <mutex>
+#include <condition_variable>
+#define _M_CEE 001
+#pragma managed(pop)
+#else
+#include <mutex>
+#include <condition_variable>
+#endif
+
+namespace VW {
+  template <typename T>
+  class ptr_queue
+  {
+  public:
+    ptr_queue(size_t max_size)
+      : max_size(max_size)
+    {}
+
+    T* pop()
```

Comment:
Why pointer-not-reference?

Reply:
A reference implies the object is still managed by the pool, a pointer makes it clearer that is now free and must be returned